From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Mon, 10 Feb 2025 16:42:39 +0800
Subject: [PATCH] Luminol: Skip collision check if the block is not belong to
 current tick region

A temporary fix of these issues:
https://github.com/PaperMC/Folia/issues/311

Due to that mojang won't ignore the blocks the entity passed by, and sometimes the old position may not be updated and it points to a block that we don't own it, and it will access it during applyEffectsFromBlocks.So the best way to fix that is to skip the collision check of that block because it was already not owned by us and it's not necessary to check it

Related to mojang's bug fix:
https://bugs.mojang.com/browse/MC-92875

diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 1d3f85189fff304b44e177544f6b83302dc880f7..2ba8eca49d0df59b3d6475114b181e56b0a8d0df 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -1672,6 +1672,11 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                         return;
                     }
 
+                    // Luminol start - Try fixing folia's old position issue
+                    if(!ca.spottedleaf.moonrise.common.util.TickThread.isTickThreadFor(this.level, blockPos) && one.tranic.sewlia.config.mod.fix.OldPosition.value)
+                        continue;
+                    // Luminol end
+
                     BlockState blockState = this.level().getBlockState(blockPos);
                     if (!blockState.isAir() && set.add(blockPos.asLong())) {
                         try {
